dali/test/python/test_operator_decoders_audio.py

```diff
@@ -152,3 +152,27 @@ def test_audio_decoder_alias():
             for quality in [0, 50, 100]:
                 for dtype in [types.INT16, types.INT32, types.FLOAT]:
                     yield check_audio_decoder_alias, sample_rate, downmix, quality, dtype
+
+def check_audio_decoder_correctness(fmt, dtype):
+  batch_size = 16
+  @pipeline_def(batch_size=batch_size, device_id=0, num_threads=4)
+  def numpy_reader_pipe(fnames, device="cpu"):
+      data = fn.readers.numpy(device=device, files=fnames)
+      return data
+
+  @pipeline_def(batch_size=batch_size, device_id=0, num_threads=4)
+  def audio_decoder_pipe(fnames, dtype, downmix=True):
+      encoded, _ = fn.readers.file(files=fnames)
+      decoded, _ = fn.decoders.audio(encoded, dtype=dtype, downmix=downmix)
+      return decoded
+
+  audio_files = get_files(f'db/audio/{fmt}', fmt)
+  npy_files = [os.path.splitext(fpath)[0] + '.npy' for fpath in audio_files]
+  new_pipe = audio_decoder_pipe(audio_files, dtype)
+  npy_pipe = numpy_reader_pipe(npy_files)
+  compare_pipelines(new_pipe, npy_pipe, batch_size, 10, eps=1)
```

Comment:
Note: OGG decoding had a few samples differing by 1. (Comparing our audio decoder and what we used to build the reference, sox). Wav and flac were matching exactly